blog/2023/06/PSA-dont-use-threadid.md

```diff
@@ -0,0 +1,176 @@
++++ mintoclevel = 2 maxtoclevel = 3 title = "PSA: Stop using `states[threadid()]`" authors = "Mason Protter, Valentin Churavy, Ian Butterworth, ..." published = "XX June 2023" rss_pubdate = Date(2023, 06, XX) rss = """PSA: Stop using `states[threadid()]`""" +++
+
+# PSA: Stop using `states[threadid()]`
+Alt titles:
+- PSA: Multithreading with `states[threadid()]` is unsafe
+- PSA: Multithreading with `buffers[threadid()]` is unsafe
+- PSA: Don't assume `threadid()` is stable within a task
```

Comment:
I find this title rather misleadning, because the stability of `threadid()` is not the actual problem here. The problem is race conditions in updating the buffer at a specific `threadid()`. 

E.g. in the single threaded example in the blogpost, the `threadid()` is perfectly stable, it's always `1`, but the problem persists.

Reply:
A technically correct (but horrible UX) title could be

> PSA: Don't use non-threadsafe, volatile, shared state to attempt to coordinate concurrent task execution

Reply:
What I want is to catch the attention of people who are only dimly aware of this stuff and what it means.

Reply:
"You are doing it wrong! Common misconceptions about threadid() and nthreads()"

Reply:
The last part of that is good

Reply:
From @vchuravy's comment below, with a slight tweak:
```
PSA: Thread-local state is no longer recommended; Common misconceptions about threadid() and nthreads()
```

Reply:
okay I've changed the title-in-progress to that one. I'm still not really convinced it's any better since people don't really know what that means, but I also don't have strong feelings about it and the old title was also not perfect.